comms/dht/src/proto/mod.rs

```diff
@@ -88,3 +96,42 @@ impl fmt::Display for dht::JoinMessage {
         )
     }
 }
+
+//---------------------------------- Rpc Message Conversions --------------------------------------------//
+
+impl From<Peer> for rpc::Peer {
+    fn from(peer: Peer) -> Self {
+        rpc::Peer {
+            public_key: peer.public_key.to_vec(),
+            addresses: peer
+                .addresses
+                .addresses
+                .iter()
+                .map(|addr| addr.address.to_string())
+                .collect(),
+            peer_features: peer.features.bits(),
+        }
+    }
+}
+
+impl rpc::Peer {
+    pub fn to_native_peer(&self) -> Option<Peer> {
```

Comment:
Not a job for a try_into impl?

Reply:
🤦 You're right! I think I wanted to `impl TryFrom<rpc::Peer> for Peer` which would obviously not compile